akka-docs/src/main/paradox/typed/style-guide.md

```diff
@@ -295,3 +300,151 @@ Scala
 
 Java
 :  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #public-private-messages-2 }
+
+@@@ div {.group-java}
+
+### Singleton messages
+
+For messages without parameters the `enum` singleton pattern is recommended:
+
+Java
+:  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #message-enum }
+
+In the `ReceiveBuilder` it can be matched in same way as other messages:
+
+Java
+:  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #message-enum-match }
+
+@@@
+
+@@@ div {.group-java}
+
+## Lamdas vs. method references
+
+It's recommended to keep the message matching in with the `ReceiveBuilder` as short and clean as possible
+and delegate to methods. That improves readability and it's easy to navigate the methods with an IDE.
+
+The delegation can be with lambdas or [method references](https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html).
+
+Example of delegation using a lambda:
+
+Java
+:  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #on-message-lambda }
+
+When possible it's preferred to use method references instead of lambdas. The benefit is less verbosity and
+in some cases it can actually give better type inference.
+
+Java
+:  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #on-message-method-ref }
+
+`this::onGetValue` is a method reference in above example. It corresponds to `command -> onGetValue(command)`.
+
+If you are using IntelliJ IDEA it has support for converting lambdas to method references.
+
+More important than the choice between lambdas or method references is to avoid lambdas with a large block of code.
+An anti-pattern would be to inline all message handling inside the lambdas like this:
+
+Java
+:  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #on-message-lambda-anti }
+
+In a real application it would often be more than 3 lines for each message.
+It's not only making it more difficult to get an overview of the message matching, but compiler errors related
+to lambdas can sometimes be difficult to understand.
+
+Ideally, lambdas should be written in one line of code. Two lines can be ok, but three is probably too much.
+Also, don't use braces and return statements in one-line lambda bodies.
+
+@@@
+
+@@@ div {.group-scala}
+
+## Partial vs. total Function
```

Comment:
Same as `vs.` thought above.

Reply:
bit funky with entries in the TOC that disappears when you switch languages, can we make this one and the ask one L3 headers instead?

Reply:
what's funky about it? it works, and also I don't think normal users switch back and forth